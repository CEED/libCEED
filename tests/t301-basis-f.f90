!-----------------------------------------------------------------------
      program test

      include 'ceedf.h'

      integer ceed,err
      integer b,i,u,v

      integer dimn,p1d,q1d,length
      integer*8 uoffset,voffset
      parameter(dimn   = 2)
      parameter(p1d    = 4)
      parameter(q1d    = 4)
      parameter(length = q1d**dimn)

      real*8 uu(length)
      real*8 vv(length)

      character arg*32

      call getarg(1,arg)
      call ceedinit(trim(arg)//char(0),ceed,err)

      call ceedvectorcreate(ceed,length,u,err)
      call ceedvectorcreate(ceed,length,v,err)

      do i=1,length
        uu(i)=1.0
      enddo
      uoffset=0
      call ceedvectorsetarray(u,ceed_mem_host,ceed_use_pointer,uu,uoffset,err)

      call ceedbasiscreatetensorh1lagrange(ceed,dimn,1,p1d,q1d,&
     & ceed_gauss_lobatto,b,err)

      call ceedbasisapply(b,1,ceed_notranspose,ceed_eval_interp,u,v,err)

      call ceedvectorgetarrayread(v,ceed_mem_host,vv,voffset,err)
      do i=1,length
        if (abs(vv(i+voffset)-1.) > 1.D-15) then
          write(*,*) 'v(',i,'=',vv(i+voffset),' not eqaul to 1.0'
        endif
      enddo
      call ceedvectorrestorearrayread(v,vv,voffset,err)

      call ceedbasisdestroy(b,err)
      call ceeddestroy(ceed,err)

      end
!-----------------------------------------------------------------------
