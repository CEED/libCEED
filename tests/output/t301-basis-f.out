 -2.00000000
 -3.00000000
 -2.00000000
  0.33333333
 -5.00000000
  2.00000000
  0.33333333
  0.40000000
 -4.00000000
  0.33333333
 -0.20000000
 -0.50000000
  1.50000000
  1.66666667
  1.60000000
